src/backend/distributed/commands/vacuum.c

```diff
@@ -210,20 +212,25 @@ VacuumTaskList(Oid relationId, CitusVacuumParams vacuumParams, List *vacuumColum
 	foreach_ptr(shardInterval, shardIntervalList)
 	{
 		uint64 shardId = shardInterval->shardId;
+		char *shardRelationName = pstrdup(relationName);
+
+		/* build shard relation name */
+		AppendShardIdToName(&shardRelationName, shardId);
 
-		char *shardName = pstrdup(tableName);
-		AppendShardIdToName(&shardName, shardInterval->shardId);
-		shardName = quote_qualified_identifier(schemaName, shardName);
+		char *quotedShardName = quote_qualified_identifier(schemaName, shardRelationName);
 
-		vacuumString->len = vacuumPrefixLen;
-		appendStringInfoString(vacuumString, shardName);
-		appendStringInfoString(vacuumString, columnNames);
+		/* copy base vacuum string and build the shard specific command */
```

Comment:
One very minor thing that confused me before approving the PR (and this is not related to this PR, the old code acts like that) is the following. The white spaces between `vacuumStringPrefix`, `quotedShardName` and `columnNames` are implicitly added in `DeparseVacuumStmtPrefix` and `DeparseVacuumColumnNames`.

I don't think we should change it now, but noted here as an outcome of my code-review for reference.